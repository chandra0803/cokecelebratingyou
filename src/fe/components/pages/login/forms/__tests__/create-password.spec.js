/*global jest*/

jest.mock( 'react-redux' );
// jest.mock( 'react-router-dom' );
// jest.mock( '../../../shared/locale-string' );
import { mount, shallow } from 'enzyme';

import LoaderButton from '../../../../shared/loader-button/loader-button';
import React from 'react';
import renderer from 'react-test-renderer';
import CreatePassword from '../create-password';

const defaultProps = {
    login: {

        locales: [

                {
                    code: 'zh_CN',
                    label: 'Chinese [Simplified]'
                },

                {
                    code: 'zh_TW',
                    label: 'Chinese [Traditional]'
                },

                {
                    code: 'nl_NL',
                    label: 'Dutch [Netherlands]'
                },

                {
                    code: 'en_GB',
                    label: 'English [British]'
                },

                {
                    code: 'en_US',
                    label: 'English [U.S]'
                },

                {
                    code: 'fr_CA',
                    label: 'French [Canadian]'
                },

                {
                    code: 'fr_FR',
                    label: 'French [Europe]'
                },

                {
                    code: 'de_DE',
                    label: 'German [Germany]'
                },

                {
                    code: 'it_IT',
                    label: 'Italian [Italy]'
                },

                {
                    code: 'ja_JP',
                    label: 'Japanese [Japan]'
                },

                {
                    code: 'ar_AE',
                    label: 'Klingon'
                },

                {
                    code: 'ko_KR',
                    label: 'Korean [South Korea]'
                },

                {
                    code: 'pl_PL',
                    label: 'Polish [Poland]'
                },

                {
                    code: 'pt_BR',
                    label: 'Portuguese [Brazil]'
                },

                {
                    code: 'ru_RU',
                    label: 'Russian [Russia]'
                },

                {
                    code: 'es_MX',
                    label: 'Spanish [Latin America]'
                },

                {
                    code: 'es_ES',
                    label: 'Spanish [Spain]'
                },

                {
                    code: 'th_TH',
                    label: 'Thai [Thailand]'
                },

                {
                    code: 'tr_TR',
                    label: 'Turkish [Turkey]'
                },

                {
                    code: 'vi_VN',
                    label: 'Vietnamese [Vietnam]'
                },

        ],

    },
    forgotId: {

    },
    forgotPass: {

    },
    content: [
        {
            key: 'REGISTER',
            code: 'login.loginpage',
            content: 'Register'
        },
        {
            key: 'TITLE',
            code: 'login.loginpage',
            content: 'Welcome'
        },
        {
            key: 'FORGOT_PASSWORD',
            code: 'login.loginpage',
            content: 'Forgot Password'
        },
        {
            key: 'FORGOT_LOGIN_ID',
            code: 'login.loginpage',
            content: 'Forgot Login ID'
        },
        {
            key: 'REGISTER_NEW_ACCOUNT',
            code: 'login.loginpage',
            content: 'Register a new account'
        },
        {
            key: 'LOG_IN',
            code: 'login.loginpage',
            content: 'Log In'
        },
        {
            key: 'NO_LOGIN_ID_OR_PASSWORD',
            code: 'login.loginpage',
            content: 'No Login ID or Password?'
        },
        {
            key: 'PASSWORD',
            code: 'login.loginpage',
            content: 'Password'
        },
        {
            key: 'I_FORGOT_PASSWORD',
            code: 'login.loginpage',
            content: 'I forgot my password'
        },
        {
            key: 'I_FORGOT_LOGIN_ID',
            code: 'login.loginpage',
            content: 'I forgot my login ID'
        },
        {
            key: 'USERNAME',
            code: 'login.loginpage',
            content: 'Login ID'
        },
        {
            key: 'INTRO_PARA_ONE',
            code: 'login.loginpage',
            content: 'This is your one-stop shop for all of your program information. Check back often as information changes regularly.'
        },
        {
            key: 'INTRO_PARA_TWO',
            code: 'login.loginpage',
            content: '???login.loginpage.INTRO_PARA_TWO???'
        },
        {
            key: 'LOGIN_ID_REQ',
            code: 'login.errors',
            content: 'Login ID is required.'
        },
        {
            key: 'PASSWORD_REQ',
            code: 'login.errors',
            content: 'Password is required.'
        },
        {
            key: 'COPYRIGHT_TEXT',
            code: 'system.general',
            content: 'Copyright &copy;2017 BI WORLDWIDE&trade;. All rights reserved.'
        },
        {
            key: 'CONTACT_US',
            code: 'system.general',
            content: 'Contact Us'
        },
        {
            key: 'T&C',
            code: 'system.general',
            content: 'T&Cs'
        },
        {
            key: 'PRIVACY_POLICY',
            code: 'system.general',
            content: 'Privacy Policy'
        }
    ],            selfEnrollment: false,
    contactUsEmailConfirmation: false,
    strutsToken: '570210087a24f665d94727351f1c6597',
    verification: {
        activation: {
            userName: 'bhd-001',

        },
        token: {
            token: ''
        }
    },
    passConfirm: {
        ignoreValidation: false,
        minRules: 3,
        minRulesLabel: 'Must be complex.  Match at least 3:',
        passwordRules: {
            isLowerCaseRequired: true,
            isNumberRequired: true,
            isSpecialCharRequired: true,
            isUpperCaseRequired: true,
            minLength: 8
        },
        passwordLabels: {}
    }

};

Object.defineProperty( window.location, 'href', {
  writable: true,
  value: 'http://localhost:8001/g6bb8/login.do?userToken=H810x421'
} );
// const getString = ( key ) => 'Mock Data: ' + key;

it( 'renders the dom like the snapshot when the props have default values.', () => {
    const props = {
        ...defaultProps
    };

    const component = renderer.create( <CreatePassword { ...props } /> );
    const tree = component.toJSON();

    expect( tree ).toMatchSnapshot();
} );
it ( 'disables the button when a password has not been entered', () => {
    const props = {
        ...defaultProps
    };

    const tree = mount( <CreatePassword { ...props } /> );

    const button = tree.find( LoaderButton );

    expect ( button.props().disabled ).toBe( true );

    tree.setState( { password1: '' } );

    expect ( button.props().disabled ).toBe( true );

} );
it ( 'enables the button when passwords are entered.', () => {
    const props = {
        ...defaultProps
    };

    const tree = mount( <CreatePassword { ...props } /> );

    const button = tree.find( LoaderButton );

    expect ( button.props().disabled ).toBe( true );
    const input = tree.find( 'input[name="password1"]' );
    input.simulate( 'change', { target: { name: input.props().name, value: '!Q@W3e4r' } } );

    expect ( button.props().disabled ).toBe( false );

} );
it ( 'executes change form prop when the top back to login link is clicked.', () => {
    const props = {
        ...defaultProps,
        handleFormChange: jest.fn()
    };

    const tree = mount( <CreatePassword { ...props } /> );

    const link = tree.find( 'a' ).first();

    link.simulate( 'click' );

    expect ( props.handleFormChange.mock.calls.length ).toBe( 1 );
} );
it ( 'shows a error alert when there is a error message.', () => {
    const login = {
        login: {
            ...defaultProps.login,
            message: {
                error: 'error'
            }
        }
    };
    const props = {
        ...defaultProps,
        ...login
    };

    const tree = mount( <CreatePassword { ...props } /> );

    const alerts = tree.find( '.alert-error' );

    expect ( alerts.length ).toBe( 1 );
} );
it ( 'shows error alerts when there are error messages.', () => {
    const login = {
        login: {
            ...defaultProps.login,
            messages:
                [ {
                    error: 'error'
                },
                {
                    error: 'error'
                } ]
        }
    };
    const props = {
        ...defaultProps,
        ...login
    };

    const tree = mount( <CreatePassword { ...props } /> );

    const alerts = tree.find( '.alert-error' );

    expect ( alerts.length ).toBe( 2 );
} );
it ( 'shows a success alert when there is a success message.', () => {
    const login = {
        login: {
            ...defaultProps.login,
            message: {
                success: 'success'
            }
        }
    };
    const props = {
        ...defaultProps,
        ...login
    };

    const tree = mount( <CreatePassword { ...props } /> );

    const alerts = tree.find( '.alert-success' );

    expect ( alerts.length ).toBe( 1 );
} );
it ( 'executes submitPass prop when the user presses enter in the password input.', () => {
    const props = {
        ...defaultProps,
        submitPass: jest.fn()
    };

    const tree = mount( <CreatePassword { ...props } /> );

    tree.setState( { password1: 'password', disabled: false } );

    const pass1 = tree.find( 'input[name="password1"]' );

    pass1.simulate( 'keyPress', { charCode: 13 } );

    expect ( props.submitPass.mock.calls.length ).toBe( 1 );
} );
it ( 'does not execute submitPass prop when the user presses key other than enter in the password input with matching passwords.', () => {
    const props = {
        ...defaultProps,
        submitPass: jest.fn()
    };

    const tree = mount( <CreatePassword { ...props } /> );

    tree.setState( { password1: 'password' } );

    const pass1 = tree.find( 'input[name="password1"]' );

    pass1.simulate( 'keyPress', { charCode: 12 } );

    expect ( props.submitPass.mock.calls.length ).toBe( 0 );
} );
it ( 'does not execute submitPass prop when the user presses enter in the password input no password.', () => {
    const props = {
        ...defaultProps,
        submitPass: jest.fn()
    };

    const tree = mount( <CreatePassword { ...props } /> );
    expect( tree.state( 'password1' ) ).toBe( '' );
    const pass1 = tree.find( 'input[name="password1"]' );

    pass1.simulate( 'keyPress', { charCode: 13 } );

    expect ( props.submitPass.mock.calls.length ).toBe( 0 );
} );
it ( 'does not execute submitPass prop when the user presses enter in the password input and disabled.', () => {
    const props = {
        ...defaultProps,
        submitPass: jest.fn()
    };

    const tree = mount( <CreatePassword { ...props } /> );
    tree.setState( { password1: 'password', disabled: true } );
    const pass1 = tree.find( 'input[name="password1"]' );

    pass1.simulate( 'keyPress', { charCode: 13 } );

    expect ( props.submitPass.mock.calls.length ).toBe( 0 );
} );
it ( 'changes passsword in state when the user changes something in the password1 input ', () => {
    const props = {
        ...defaultProps
    };

    const tree = mount( <CreatePassword { ...props } /> );

    const input = tree.find( 'input[name="password1"]' );

    input.simulate( 'change', { target: { name: input.props().name, value: 'test' } } );

    expect ( tree.state( 'password1' ) ).toBe( 'test' );
} );
it ( 'executes modal prop when the help link is clicked.', () => {
    const props = {
        ...defaultProps,
        toggleModalDisplay: jest.fn()
    };

    const tree = mount( <CreatePassword { ...props } /> );

    const link3 = tree.find( '.help a' );

    link3.simulate( 'click' );

    expect ( props.toggleModalDisplay.mock.calls.length ).toBe( 1 );
} );
it ( 'executes submitPass prop when the user presses enter in the password input.', () => {
    const rules = {
        passConfirm: {
            passwordRules: {
                isLowerCaseRequired: false,
                isNumberRequired: false,
                isSpecialCharRequired: false,
                isUpperCaseRequired: false,
                minLength: 0
            },
            passwordLabels: {}
        }
    };
    const props = {
        ...defaultProps,
        ...rules,
        submitPass: jest.fn()
    };

    const tree = mount( <CreatePassword { ...props } /> );

    tree.instance().validate( '!Q@W3e4r' );
    tree.setState( { password1: 'password', disabled: false } );
    const pass1 = tree.find( 'input[name="password1"]' );

    pass1.simulate( 'keyPress', { charCode: 13 } );

    expect ( props.submitPass.mock.calls.length ).toBe( 1 );
} );
