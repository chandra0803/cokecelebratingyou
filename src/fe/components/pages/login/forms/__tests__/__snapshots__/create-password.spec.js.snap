// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the dom like the snapshot when the props have default values. 1`] = `
<div
  className="narrow-form-page login"
>
  <form
    action={undefined}
    className="narrow-form"
    method={undefined}
  >
    <p>
      <a
        onClick={[Function]}
        tabIndex="5"
      >
        <span>
          <i
            className="icon-arrow-1-circle-left"
          />
           
        </span>
      </a>
    </p>
    <h3
      className="no-bottom"
    />
    <p
      className="loginID"
    >
      undefined: bhd-001
    </p>
    <div
      className="password-helper-input"
    >
      <div
        className="password-input"
      >
        <div
          className="input-wrap"
        >
          <label
            className="control-label"
            htmlFor="password1"
          />
          <span
            className="eye"
            onClick={[Function]}
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,3.9C4.2,3.9,0,9.6,0,12s4.2,8.1,12,8.1s12-5.8,12-8.1S19.8,3.9,12,3.9z M12,18.7c-7.2,0-10.6-5.4-10.6-6.7 S4.8,5.3,12,5.3s10.6,5.4,10.6,6.7S19.2,18.7,12,18.7z"
              />
              <path
                d="M12,7c-2.8,0-5,2.2-5,5s2.2,5,5,5s5-2.2,5-5S14.8,7,12,7z M12,15.6c-2,0-3.6-1.6-3.6-3.6c0-2,1.6-3.6,3.6-3.6 c2,0,3.6,1.6,3.6,3.6C15.6,14,14,15.6,12,15.6z"
              />
            </svg>
          </span>
          <input
            autoFocus={true}
            className="gq-input"
            disabled={undefined}
            name="password1"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            tabIndex="1"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="password-rules-container hide"
      >
        <div
          className="rule"
        >
          <div
            className="hide"
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
              />
              <path
                d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
              />
            </svg>
          </div>
          <div
            className="hide"
          >
            <span>
              <svg
                className="icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14.6,8.3l-2.7,2.7L9.3,8.3C9,8,8.5,8,8.2,8.3c-0.3,0.3-0.4,0.8,0,1.1l2.7,2.7l-2.5,2.5C8,15,8,15.5,8.3,15.8 c0.3,0.3,0.8,0.4,1.1,0l2.7-2.7l2.7,2.7c0.3,0.3,0.8,0.4,1.1,0c0.3-0.3,0.4-0.8,0-1.1L13.3,12L16,9.3c0.3-0.3,0.4-0.8,0-1.1 C15.4,8,14.9,8,14.6,8.3z M3.5,3.5c-4.7,4.7-4.7,12.3,0,17s12.3,4.7,17,0s4.7-12.3,0-17S8.2-1.2,3.5,3.5z M19.4,19.4 c-4.1,4.1-10.7,4.1-14.8,0S0.5,8.7,4.6,4.6s10.7-4.1,14.8,0S23.5,15.3,19.4,19.4z"
                />
              </svg>
            </span>
          </div>
          <span />
        </div>
        <div
          className="rule"
        >
          <div
            className="hide"
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
              />
              <path
                d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
              />
            </svg>
          </div>
          <div
            className="rule-icon close-circle-icon"
          >
            <span>
              <svg
                className="icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14.6,8.3l-2.7,2.7L9.3,8.3C9,8,8.5,8,8.2,8.3c-0.3,0.3-0.4,0.8,0,1.1l2.7,2.7l-2.5,2.5C8,15,8,15.5,8.3,15.8 c0.3,0.3,0.8,0.4,1.1,0l2.7-2.7l2.7,2.7c0.3,0.3,0.8,0.4,1.1,0c0.3-0.3,0.4-0.8,0-1.1L13.3,12L16,9.3c0.3-0.3,0.4-0.8,0-1.1 C15.4,8,14.9,8,14.6,8.3z M3.5,3.5c-4.7,4.7-4.7,12.3,0,17s12.3,4.7,17,0s4.7-12.3,0-17S8.2-1.2,3.5,3.5z M19.4,19.4 c-4.1,4.1-10.7,4.1-14.8,0S0.5,8.7,4.6,4.6s10.7-4.1,14.8,0S23.5,15.3,19.4,19.4z"
                />
              </svg>
            </span>
          </div>
          <span>
            Must be complex.  Match at least 3:
          </span>
        </div>
        <div
          className="rule child-label"
        >
          <div
            className="hide"
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
              />
              <path
                d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
              />
            </svg>
          </div>
          <div
            className="hide"
          >
            <span
              className="circle"
            />
          </div>
          <span />
        </div>
        <div
          className="rule child-label"
        >
          <div
            className="hide"
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
              />
              <path
                d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
              />
            </svg>
          </div>
          <div
            className="hide"
          >
            <span
              className="circle"
            />
          </div>
          <span />
        </div>
        <div
          className="rule child-label"
        >
          <div
            className="hide"
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
              />
              <path
                d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
              />
            </svg>
          </div>
          <div
            className="hide"
          >
            <span
              className="circle"
            />
          </div>
          <span />
        </div>
        <div
          className="rule child-label"
        >
          <div
            className="hide"
          >
            <svg
              className="icon"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,24C5.4,24,0,18.6,0,12S5.4,0,12,0s12,5.4,12,12S18.6,24,12,24z M12,1.5C6.2,1.5,1.5,6.2,1.5,12S6.2,22.5,12,22.5 S22.5,17.8,22.5,12S17.8,1.5,12,1.5z"
              />
              <path
                d="M10.6,16.6c-0.2,0-0.4-0.1-0.5-0.2l-4.4-4.4c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l3.8,3.8l6.7-7.2 c0.3-0.3,0.8-0.3,1.1,0c0.3,0.3,0.3,0.8,0,1.1l-7.2,7.7C11,16.6,10.8,16.6,10.6,16.6C10.6,16.6,10.6,16.6,10.6,16.6z"
              />
            </svg>
          </div>
          <div
            className="hide"
          >
            <span
              className="circle"
            />
          </div>
          <span />
        </div>
      </div>
    </div>
    <button
      className="gq-btn disabled btn btn-block btn-primary form-action-btn"
      onClick={[Function]}
      tabIndex={3}
      type="button"
    />
    <p
      className="help"
    >
      <a
        onClick={[Function]}
        tabIndex="3"
      >
        <span />
      </a>
    </p>
  </form>
  <div
    className="theme-image"
  />
</div>
`;
